### PR TITLE
docs: simplify README.md

### PR DESCRIPTION
**Pull request by LukeWinikates**
* move releasing instructions into their own file
* remove table discussing SDKs not in this repo
* remove examples in README in favor of code examples
* add internal SDK metrics to this README

### CODE REVIEW

**Comment by LukeWinikates** [comment]
see #106

**Review by keep94 — COMMENTED** [review]
It seems you are removing all the SDK examples from the readme documentation. I believe this is the direction we want to go, but I don't remember seeing example code in your other PR for histograms for instance.

**Comment by LukeWinikates** [comment]
> It seems you are removing all the SDK examples from the readme documentation. I believe this is the direction we want to go, but I don't remember seeing example code in your other PR for histograms for instance.

Thanks Travis - yep, this PR removes the examples from the README. It looks like I did convert all of the readme examples into godoc examples, so spans, histograms, metrics, and delta counters are all included in the test file added by this PR.

**Review by keep94 — APPROVED** [review]
LGTM

**Comment by keep94** [comment]
Hi shavidissa, we are planning to move all the example code over to pkg.go.dev and add a link to pkg.go.dev from the readme.  For you to see the changes on pkg.go.dev, we'd have to do a release, but if you download the godoc tool on your machine and checkout this PR, you can get a fairly accurate preview of what the page on pkg.go.dev will look like.  I can show you how to do this if you'd like.

File: README.md [hunk 1]
```diff
@@ -4,301 +4,41 @@
...
-Wavefront by VMware Go SDK lets you send raw data from your Go application to Wavefront using a `Sender` interface. The data is then stored as metrics, histograms, and trace data. This SDK is also called the Wavefront Sender SDK for Go. 
+Wavefront by VMware Go SDK lets you send raw data from your Go application to Wavefront using a `Sender` interface.
+The data is then stored as metrics, histograms, and trace data. This SDK is also called the Wavefront Sender SDK for Go. 
 
-Although this library is mostly used by the other Wavefront Go SDKs to send data to Wavefront, you can also use this SDK directly. For example, you can send data directly from a data store or CSV file to Wavefront.
+Although this library is mostly used by the other Wavefront Go SDKs to send data to Wavefront, 
+you can also use this SDK directly. For example, you can send data directly from a data store or CSV file to Wavefront.
 
-**Before you start implementing, let us make sure you are using the correct SDK!**
+For detailed documentation of SDK types and functions, see [pkg.go.dev](https://pkg.go.dev/github.com/wavefronthq/wavefront-sdk-go/senders)
```
Comment (shavidissa) [inline README.md:19]: To lean more on how to send data, the SDK types, and functions, see [pkg.go.dev documentation](https://pkg.go.dev/github.com/wavefronthq/wavefront-sdk-go/senders)

File: README.md [hunk 2]
```diff
@@ -4,301 +4,41 @@
 [![Go Reference](https://pkg.go.dev/badge/github.com/wavefronthq/wavefront-sdk-go.svg)](https://pkg.go.dev/github.com/wavefronthq/wavefront-sdk-go)
 [![Go Report Card](https://goreportcard.com/badge/github.com/wavefronthq/wavefront-sdk-go)](https://goreportcard.com/report/github.com/wavefronthq/wavefront-sdk-go)
 
-## Table of Content
-* [Prerequisites](#Prerequisites)
-* [Set Up a Sender](#set-up-a-sender)
-* [Send Data to Wavefront](#send-data-to-wavefront)
-* [Close the Sender](#close-the-sender)
+## Table of Contents
 * [License](#License)
```
Comment (shavidissa) [inline README.md:8]: Add the following too:

* [Internal SDK Metrics](#internal-sdk-metrics)

File: README.md [hunk 3]
```diff
@@ -307,27 +47,3 @@ sender.Close()
```
Comment (shavidissa) [inline README.md:47]: Add a point here linking to the how to release readme. :)

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+## How to Release the SDK
+
+1. Make sure that all changes for the release are merged with the master branch.
+2. From the root directory of the project in the master branch, check that the following command run cleanly.
```
Comment (shavidissa) [inline RELEASING.md:4]: From the root directory of the project in the master branch, run the following commands.

File: RELEASING.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+3. Run `go mod tidy` to make sure it doesn't introduce new changes. If it brings new changes, check those in first before doing the release.
+4. Click **Releases** in the about section to get the details on the version history.
+5. Decide what the version of the next release will be. We follow [semantic versioning](https://semver.org/).
+6. Execute `git tag -a -m 'v0.10.3' 'v0.10.3'` replacing v0.10.3 with the new version.
+7. Push the tag up to Github `git push upstream v0.10.3`.
+8. Log into Github, click on **Releases** in the about section on the right, and click on **Draft a new release.**
+9. For **version**, choose the version you decided upon in step 5.
+10. Provide a short but descriptive title for the release.
+11. Fill in the details of the release. Please copy the markdown from the previous release and follow the same format.
+12. Click on "Publish release."
```
Comment (shavidissa) [inline RELEASING.md:17]: Click **Publish release**.

**Review by shavidissa — APPROVED** [review]
Thank you for the explanation! I have a few minor comments.
LGTM after the changes are made :)

**Comment by LukeWinikates** [comment]
thanks @shavidissa!

@keep94 I also created #118 as a place to discuss whether we can make the RELEASING.md steps easier.

**Comment by LukeWinikates** [comment]
@keep94 do you think we're ok to merge this PR now? I think all the review comments are resolved

**Pull request merged**